File: csharp/hamming/Hamming.cs
using System;

public static class Hamming
{
    public static int Distance(string firstStrand, string secondStrand)
    {
        int hamming = 0;
        
        if (firstStrand.Length != secondStrand.Length)
        {
            throw new ArgumentException("The two strings must be the same length");
        }
        
        for (int i = 0; i < firstStrand.Length; i++)
        {
            if (firstStrand[i] != secondStrand[i])
            {
                hamming ++;
            }
        }
        return hamming;
    }
}

File: go/hamming/hamming.go
// Package hamming includes a function to calculate the Hamming distance of two strings
package hamming

import (
	"fmt"
)

// Distance takes two strings and returns their Hamming distance. Returns error if strings are of different lengths.
func Distance(a, b string) (int, error) {
	var hammingDistance int
	if len(a) != len(b) {
		return hammingDistance, fmt.Errorf("%s and %s are different lengths", a, b)
	}
	for i := 0; i < len(a); i++ {
		if a[i] != b[i] {
			hammingDistance++
		}
	}
	return hammingDistance, nil
}
